### ISSUE TITLE
dev-dependencies dose not support Multiple constraints dependencies

### ISSUE THREAD

**Issue by crhan** [python-poetry/poetry#1448]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: osx 10.14.6
- **Poetry version**: 0.12.17
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/crhan/d63656ad944cfcd28ed3c427e3debcc0

## Issue

It seems like dependency defined in `[tool.poetry.dev-dependencies]` section does not support [__Multiple constraints dependencies__](https://poetry.eustace.io/docs/versions/#multiple-constraints-dependencies) feature.

Output pasted blow:

```
$ poetry lock -vvv
Using virtualenv: /Users/ruohan.chen/.local/share/virtualenvs/tmp-1889ae4e3228fbb
Updating dependencies
Resolving dependencies...
   1: fact: 20191008 is 0.1.0
   1: derived: 20191008
   1: fact: 20191008 depends on pytest (^5.1)
   1: fact: 20191008 depends on pytest (<5)
   1: selecting 20191008 (0.1.0)
   1: derived: pytest (<5)
   1: conflict: 20191008 depends on pytest (^5.1)
   1: ! not pytest (^5.1) is satisfied by pytest (<5)
   1: ! which is caused by "20191008 depends on pytest (<5)"
   1: ! thus: version solving failed
   1: Version solving took 0.017 seconds.
   1: Tried 1 solutions.

[SolverProblemError]
Because 20191008 depends on both pytest (^5.1) and pytest (<5), version solving failed.

Exception trace:
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/console/commands/lock.py in handle() at line 32
   return installer.run()
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/installation/installer.py in run() at line 73
   self._do_install(local_repo)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/installation/installer.py in _do_install() at line 165
   ops = solver.solve(use_latest=self._whitelist)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/puzzle/solver.py in solve() at line 38
   packages, depths = self._solve(use_latest=use_latest)
 /Users/ruohan.chen/.local/pipx/venvs/poetry/lib/python3.7/site-packages/poetry/puzzle/solver.py in _solve() at line 180
   raise SolverProblemError(e)

lock
```

**Comment by danihodovic** [comment]
I'm facing the same issue

**Comment by sdispater** [comment]
@danihodovic Which version of Poetry do you use? And can you post the complete output of the error?

As far as I know, it is fixed in the `1.0.0` version.

**Comment by sdispater** [comment]
I will assume that this is now fixed.

@danihodovic If you still face this issue with the latest Poetry version, feel free to open a new issue.
